### DISCUSSION TITLE
CheckMATE Installation Issue: Cannot Find Delphes Header

### DISCUSSION THREAD

**Discussion by wrishik** [CheckMATE2/checkmate2#33] [General]
Hi CheckMATE team,

I have been trying to install CheckMATE for 2 days, and I am running into the same issue all the time. Here's the error message I get:

........
checking for Delphes location... ../Delphes/
checking for classes/DelphesClasses.h... no
configure: error: Cannot find DelphesClasses header
........

I have checked my Delphes installation, and indeed there is a the DelphesClasses.h header file in that specified location and smh CheckMATE cannot identify it. I am installing the latest version, i.e., 'checkmate2'. I would be really grateful if you could help me in this matter.

Thanking you.
Kind regards,
Wrishik.

**Comment by NishitaDesai** [comment]
Hi Wrishik,  can you paste your configure command and the path of where you've installed your Delphes?  Also, what OS/version are you using?

**Reply by wrishik** [reply]
Hi!
Many thanks for your response!

Here is the configure command I am using:

./configure --with-rootsys=../root-6.26.10/ROOT/ --with-delphes=../Delphes/ --with-pythia=../MG5_aMC_v2_9_7/HEPTools/pythia8/ --with-madgraph=../MG5_aMC_v2_9_7/

I am using macOS Monterey: v12.6.5

Thank you!

**Comment by NishitaDesai** [comment]
Thanks.  But that still doesn't answer my install path question.  Can you go into your Delphes directory and send me the output of pwd?  Also, try using full paths instead of relative and see if that helps.

**Reply by wrishik** [reply]
Hi,

I'm really sorry, I didn't send it previously. Here's my Delphes directory:

/Users/wrishiknaskar/Documents/HEPTools/Delphes

I tried using full paths, but I am running into the same error.

Thank you so much for you time!

**Comment by NishitaDesai** [comment]
Did you try disabling SIP (https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection) and exporting DYLD_LIBRARY_PATH to include paths to all the right libraries.

e.g. export DYLD_LIBRARY_PATH=/Users/wrishiknaskar/Documents/HEPTools/Delphes:/path/to/root/ etc.

**Comment by foice** [comment]
fyi just got the same issue on Almalinux9.  I fixed this doing all of the following:

```
export ROOT_INCLUDE_PATH=$ROOT_INCLUDE_PATH:/home/user/Delphes-3.5.0
export LIBRARY_PATH=$LIBRARY_PATH:/home/user/Delphes-3.5.0/
```
running with the configure option `--with-delphes=/home/user/Delphes-3.5.0/`
and making sure that ROOT `thisroot` was sourced from the same ROOT installation folder as the`root` command, i.e. `which root` gives `~/root/bin/root` and I made sure to have done `source ~/root/bin/thisroot.sh`
